tip-distributor/src/claim_mev_workflow.rs

```diff
@@ -32,121 +39,343 @@ pub enum ClaimMevError {
-        if !failed_transactions.is_empty() {
-            panic!("failed to send {} transactions", failed_transactions.len());
-        }
-    });
+    datapoint_info!(
+        "claim_mev_workflow-prepare_transactions",
+        ("tree_node_count", tree_node_count, i64),
+        ("tda_count", tdas.len(), i64),
+        ("claimant_count", claimants.len(), i64),
+        ("claim_status_count", claim_statuses.len(), i64),
+        ("skipped_merkle_root_count", skipped_merkle_root_count, i64),
+        ("zero_lamports_count", zero_lamports_count, i64),
+        ("already_claimed_count", already_claimed_count, i64),
+        ("below_min_rent_count", below_min_rent_count, i64),
+        ("instruction_count", instructions_len, i64),
+        ("transaction_count", transactions.len(), i64),
+        (
+            "account_fetch_latency_us",
+            account_fetch_elapsed.as_micros(),
+            i64
+        ),
+        ("latency_us", transaction_prepare_elapsed.as_micros(), i64),
+    );
+
+    info!("Sending {} tip claim transactions. {zero_lamports_count} tried sending zero lamports, {below_min_rent_count} would be below minimum rent",
+            transactions.len());
+    let mut retries = 0;
+    let mut failed_transactions = HashMap::new();
+    while !transactions.is_empty() && retries < MAX_SEND_RETRIES {
+        let transactions_len = transactions.len();
+        let send_start = Instant::now();
+        let (to_process_transactions, new_failed_transactions) =
+            sign_and_send_transactions_with_retries(
+                &keypair,
+                &rpc_client,
+                transactions,
+                MAX_RETRY_DURATION,
+            )
+            .await;
+
+        datapoint_info!(
+            "claim_mev_workflow-send_transactions",
+            ("transaction_count", transactions_len, i64),
+            (
+                "successful_transaction_count",
+                transactions_len - to_process_transactions.len(),
+                i64
+            ),
+            (
+                "remaining_transaction_count",
+                to_process_transactions.len(),
+                i64
+            ),
+            (
+                "failed_transaction_count",
+                new_failed_transactions.len(),
+                i64
+            ),
+            ("send_latency_us", send_start.elapsed().as_micros(), i64),
+        );
+        transactions = to_process_transactions;
+        failed_transactions.extend(new_failed_transactions);
+        retries += 1;
+    }
+    if !failed_transactions.is_empty() {
+        panic!(
+            "Failed to send {} mev claim transactions",
+            failed_transactions.len()
+        );
+    }
 
     Ok(())
 }
+async fn get_batched_accounts(
+    rpc_client: &RpcClient,
+    pubkeys: Vec<Pubkey>,
+) -> solana_rpc_client_api::client_error::Result<HashMap<Pubkey, Option<Account>>> {
```

Comment:
why not just `HashMap<Pubkey, Account>` and if DNE don't include the entry

Reply:
rn looking to see it panic if i have a logic error. will switch over after testing :)